src/schemas/Entity.js

```diff
@@ -33,6 +33,31 @@ export default class EntitySchema {
     return this._idAttribute;
   }
 
+  stringifySafe(input) {
+    function replacer() {
+      const stack = [];
+      const keys = [];
+
+      return (key, value) => {
+        if (stack.length > 0) {
+          const thisPos = stack.indexOf(this);
+          ~thisPos ? stack.splice(thisPos + 1) : stack.push(this);
```

Comment:
Please don't use bitwise operators. They're very unfriendly for those who are not super familiar with what they're doing and I'd like to keep normalizr easy to contribute to.